File: app/src/types/work-experience.ts
export interface WorkExperienceForm {
  id?: number;
  companyId?: number;
  company?: string;
  position?: string;
  description?: string;
  startDate?: string;
  endDate?: string;
  skills?: string[];
  achievements?: string[];
}

export interface WorkExperienceQuery extends BodyQuery {

}


export interface WorkExperienceResult {
  id: number;
  createdAt: string;
  updatedAt: string;
  deleted: number;
  userId: number;
  companyId: number;
  position: string;
  startDate: string;
  endDate: string;
  isCurrent: boolean;
  description: string;
  achievements: string[] | null;
  sort: number;
}

File: server/src/main/java/cn/net/yunlou/bole/model/entity/WorkExperience.java
package cn.net.yunlou.bole.model.entity;

import cn.net.yunlou.bole.common.BaseEntity;
import cn.net.yunlou.bole.common.handler.JsonbTypeHandler;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.fasterxml.jackson.annotation.JsonFormat;
import java.time.LocalDate;
import java.util.List;
import lombok.*;

@EqualsAndHashCode(callSuper = true)
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName(value = "t_work_experiences", autoResultMap = true, resultMap = "BaseResultMap")
public class WorkExperience extends BaseEntity {

    private Long userId;

    private Long companyId;

    private String company;

    private String position;

    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate startDate;

    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate endDate;

    private Boolean isCurrent;

    private String description;

    @TableField(typeHandler = JsonbTypeHandler.class)
    private List<String> achievements;

    private Integer sort;
}
